kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -138,48 +138,37 @@ grCollectionsPanel.directive('grDropIntoCollection',
     ['$timeout', '$parse', 'vndMimeTypes', 'collections',
     function($timeout, $parse, vndMimeTypes, collections) {
 
-    const className = 'collection-drop';
-    const classDrag = 'collection-drop--drag-over';
-    const classComplete = 'collection-drop--complete';
-    const classSaving = 'collection-drop--saving';
+    const classOver = 'collection-drop-drag-over';
 
     return {
         restrict: 'A',
         link: function(scope, element, attrs) {
             const collectionPath = $parse(attrs.grDropIntoCollection)(scope);
-            element.addClass(className);
 
             element.on('drop', jqEv => {
                 const dt = jqEv.originalEvent.dataTransfer;
                 const gridImagesData = dt.getData(vndMimeTypes.get('gridImagesData'));
                 const gridImageData = dt.getData(vndMimeTypes.get('gridImageData'));
 
-                if (gridImagesData !== '' && gridImageData !== '') {
-                    // TODO: potentially add some UI feedback on adding to collection
+                if (gridImagesData !== '' || gridImageData !== '') {
                     const imagesData = gridImagesData !== '' ?
                         JSON.parse(gridImagesData) : [JSON.parse(gridImageData)];
 
-                    const imageIds = imagesData.map(imageJson => imageJson.id);
-
-                    // TODO: Find a better way of dealing with this state than classnames
-                    element.addClass(classSaving);
+                    const imageIds = imagesData.map(imageJson => imageJson.data.id);
+                    scope.dropIntoCollectionSaving = true;
```

Comment:
Watch out with scope inheritance (just investigated a Composer bug due to use of scope in a non-scope-isolated directive with Rob yesterday). What scope is this on and does it risk to leak to siblings or children also observing it?